Assets/Scripts/Screeps3D/Rooms/RoomUnpacker.cs

```diff
@@ -135,10 +140,42 @@ private void UnpackFlags(JSONObject roomData)
                 var dataArray = flagStr.Split('~');
                 if (dataArray.Length < 5)
                     continue;
+                flagList.Add(dataArray[0]);
                 var flag = ObjectManager.Instance.GetFlag(dataArray);
                 flag.FlagDelta(dataArray, _room);
                 _room.Objects[flag.Name] = flag;
             }
+            
+            // process existing flag deltas
+            _removeList.Clear();
+            foreach (var kvp in _room.Objects.Values.OfType<Flag>().ToList())
+            {
+                
+                var id = kvp.Id;
+                var flagObject = kvp;
+
+                if (flagList.Contains(id))
+                {
+                    activeFlagList.Add(id);
```

Comment:
Instead of adding it to a flagList we can just check if flagStrings contains the flag.Name

like it does for roomobjects https://github.com/thmsndk/Screeps3D/blob/e2e906a5c189c06b30d94c7a9fbb9185fae206e6/Assets/Scripts/Screeps3D/Rooms/RoomUnpacker.cs#L60

Reply:
The issue with Flags was a pain as they were not objects, instead it was a String and had to be split.  Eg something like `Flag1~10~10~20~24|Flag2~12~10~22~21`  So they all had to be broken down then iterated.    I wish they were not strings as it would have been a lot cleaner

Reply:
But it was already split and stuffed into flagStrings as an array of flagNames
https://github.com/thmsndk/Screeps3D/blob/e2e906a5c189c06b30d94c7a9fbb9185fae206e6/Assets/Scripts/Screeps3D/Rooms/RoomUnpacker.cs#L132

Could we not just do a simple `flagStrings.Containers(flagName)` ? I might be too tired to understand why we need the id and the seperate list at the moment :)

the key of room objects are the name of flags e.g. ` _room.Objects[flag.Name] = flag;`

`flagStrings` and `activeFlagList` are the same thing, are they not? one is a list of strings, the other is an array of strings both being the flagname?

Reply:
Close :)

flagsData looks like "swarm_W3N7s~4~9~25~25|intel_nsa~4~9~25~25|control_W3N7c~4~9~25~25"
flagStrings would be and array of strings ["swarm_W3N7s~4~9~25~25", "intel_nsa~4~9~25~25", "control_W3N7c~4~9~25~25" ]

Then next flagsStrings is split into 5 and stuffed into an array called `dataArray`  
https://github.com/thmsndk/Screeps3D/blob/e2e906a5c189c06b30d94c7a9fbb9185fae206e6/Assets/Scripts/Screeps3D/Rooms/RoomUnpacker.cs#L135

Since there was no scope on this `dataArray` that dataArray[0] contained the flagName, i created the list to store the flagNames after they were broken out of the original string.  Then I can use it later to determine if that flagName was stale.